### PR TITLE
Avoid race conditions in the mo_photo functions.

### PR DESCRIPTION
**Pull request by odiazib**
We were encountering an FPE in the pm-gpu configuration of eamxx in [PR 3013](https://github.com/E3SM-Project/scream/pull/3013). To fix this issue, we made the necessary changes to avoid race conditions. However, the mo_photo function will need further modifications to fully utilize parallelism from GPUs.

### CODE REVIEW

**Comment by jaelynlitz** [comment]
```
497/625 Test #497: validate_stand_synthetic_table_photo_ts_355 ........................................***Failed    0.47 sec
photo
L1 767.1368211200008
L2 142.20983065035222
Linf 69.3308063
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.6999999985500835e-06
o1_a = 127.6893051
o2_a = 127.6893034
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 4.742858200000001
o1_a = 127.6893051
o2_a = 122.9464469
abs diff = 3.000000070585429e-08
o1_a = 10.09163902
o2_a = 10.09163899
abs diff = 2.8000000042993634e-06
o1_a = 165.5946429
o2_a = 165.5946401
abs diff = 0.05289480000000424
o1_a = 117.9955882
o2_a = 117.9426934
abs diff = 9.158482869999998
o1_a = 10.09163902
o2_a = 19.25012189
abs diff = 69.3308063
o1_a = 189.2148152
o2_a = 119.8840089
abs diff = 12.809022200000001
o1_a = 127.6893051
o2_a = 114.8802829
abs diff = 3.399999997100167e-07
o1_a = 26.92126434
o2_a = 26.921264
abs diff = 1.900000000887303e-06
o1_a = 113.6637821
o2_a = 113.6637802
abs diff = 14.533853800000003
o1_a = 127.6893051
o2_a = 113.1554513
abs diff = 20.9491941
o1_a = 10.09163902
o2_a = 31.04083312
abs diff = 43.7576608
o1_a = 189.2148152
o2_a = 145.4571544
abs diff = 1.4000000021496817e-06
o1_a = 111.7693223
o2_a = 111.7693209
abs diff = 1.0136858500000017
o1_a = 32.66883179
o2_a = 33.68251764
abs diff = 13.231452899999994
o1_a = 189.2148152
o2_a = 175.9833623
abs diff = 17.493535699999995
o1_a = 127.6893051
o2_a = 110.1957694
abs diff = 25.32844128
o1_a = 10.09163902
o2_a = 35.4200803
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 19.149427099999997
o1_a = 127.6893051
o2_a = 108.539878
abs diff = 26.56855615
o1_a = 10.09163902
o2_a = 36.66019517
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.4000000021496817e-06
o1_a = 107.1333087
o2_a = 107.1333073
abs diff = 5.500000028746399e-07
o1_a = 37.87986154
o2_a = 37.87986099
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 21.690892599999998
o1_a = 127.6893051
o2_a = 105.9984125
abs diff = 29.126629729999998
o1_a = 10.09163902
o2_a = 39.21826875
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 22.773634700000002
o1_a = 127.6893051
o2_a = 104.9156704
abs diff = 30.538292239999997
o1_a = 10.09163902
o2_a = 40.62993126
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.2999999938756446e-06
o1_a = 103.7696277
o2_a = 103.7696264
abs diff = 0.17298885000000297
o1_a = 41.79530516
o2_a = 41.96829401
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.3000000080864993e-06
o1_a = 102.553217
o2_a = 102.5532157
abs diff = 6.499999969378223e-07
o1_a = 43.18195142
o2_a = 43.18195077
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 15.2847668
o1_a = 116.9090289
o2_a = 101.6242621
abs diff = 27.485969190000002
o1_a = 16.15190959
o2_a = 43.63787878
abs diff = 22.90325709999999
o1_a = 166.311555
o2_a = 189.2148121
abs diff = 0.04405870000000789
o1_a = 101.0007594
o2_a = 100.9567007
abs diff = 3.4699605800000057
o1_a = 40.08192817
o2_a = 43.55188875
abs diff = 6.8101877
o1_a = 182.4046244
o2_a = 189.2148121
abs diff = 1.1999999998124622e-06
o1_a = 100.3220682
o2_a = 100.322067
abs diff = 6.59999997765226e-07
o1_a = 43.48069611
o2_a = 43.48069545
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 11.991766949999999
o1_a = 111.7193699
o2_a = 99.72760295
abs diff = 12.07791318
o1_a = 31.35523359
o2_a = 43.43314677
abs diff = 15.9939842
o1_a = 173.2208279
o2_a = 189.2148121
abs diff = 1.0857752800000071
o1_a = 100.2331125
o2_a = 99.14733722
abs diff = 4.212510739999999
o1_a = 39.30940423
o2_a = 43.52191497
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.229999995189246e-06
o1_a = 98.60919826
o2_a = 98.60919703
abs diff = 6.59999997765226e-07
o1_a = 43.72430542
o2_a = 43.72430476
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 8.610277389999993
o1_a = 106.7035688
o2_a = 98.09329141
abs diff = 6.60991705
o1_a = 37.55134452
o2_a = 44.16126157
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 5.219789879999993
o1_a = 102.8534912
o2_a = 97.63370132
abs diff = 3.633913619999994
o1_a = 41.25287517
o2_a = 44.88678879
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.2099999935344385e-06
o1_a = 97.22892325
o2_a = 97.22892204
abs diff = 7.000000010748408e-07
o1_a = 45.97926742
o2_a = 45.97926672
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 5.0238042599999915
o1_a = 101.8562192
o2_a = 96.83241494
abs diff = 4.279009850000001
o1_a = 43.15867651
o2_a = 47.43768636
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 3.2145397299999985
o1_a = 99.66137081
o2_a = 96.44683108
abs diff = 3.5309920700000035
o1_a = 45.67601704
o2_a = 49.20700911
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.1899999918796311e-06
o1_a = 96.04936185
o2_a = 96.04936066
abs diff = 7.999999951380232e-07
o1_a = 51.33889298
o2_a = 51.33889218
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.1799999981576548e-06
o1_a = 95.55972991
o2_a = 95.55972873
abs diff = 8.499999992750418e-07
o1_a = 53.82854337
o2_a = 53.82854252
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 3.195563219999997
o1_a = 98.20586034
o2_a = 95.01029712
abs diff = 12.019743609999999
o1_a = 44.33328357
o2_a = 56.35302718
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 0.0562566599999883
o1_a = 94.48649624
o2_a = 94.43023958
abs diff = 4.42566532
o1_a = 54.21761271
o2_a = 58.64327803
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.1599999965028474e-06
o1_a = 93.73347788
o2_a = 93.73347672
abs diff = 9.800000029258626e-07
o1_a = 61.00365755
o2_a = 61.00365657
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 4.09056837
o1_a = 97.26050969
o2_a = 93.16994132
abs diff = 17.03666236
o1_a = 46.47652073
o2_a = 63.51318309
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 2.363110849999998
o1_a = 95.28116739
o2_a = 92.91805654
abs diff = 12.648809139999997
o1_a = 53.44000991
o2_a = 66.08881905
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.1300000011260636e-06
o1_a = 92.65744635
o2_a = 92.65744522
abs diff = 1.1200000074040872e-06
o1_a = 68.76526834
o2_a = 68.76526722
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.13999999484804e-06
o1_a = 93.60636027
o2_a = 93.60635913
abs diff = 1.1799999981576548e-06
o1_a = 71.50389607
o2_a = 71.50389489
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.0162837999999965
o1_a = 93.25719057
o2_a = 94.27347437
abs diff = 7.846445520000003
o1_a = 66.33299975
o2_a = 74.17944527
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 0.009226069999996866
o1_a = 94.95720506
o2_a = 94.96643113
abs diff = 0.1371429300000102
o1_a = 76.41706903
o2_a = 76.55421196
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 1.1799999981576548e-06
o1_a = 95.91438361
o2_a = 95.91438243
abs diff = 1.4300000117373202e-06
o1_a = 84.40451554
o2_a = 84.40451411
abs diff = 3.10000001491062e-06
o1_a = 189.2148152
o2_a = 189.2148121
abs diff = 4.089076910000003
o1_a = 93.38085625
o2_a = 97.46993316
abs diff = 25.250777279999994
o1_a = 70.37478769
o2_a = 95.62556497
abs diff = 8.792022199999991
o1_a = 189.2148152
o2_a = 198.0068374
abs diff = 1.7379515900000087
o1_a = 97.52694512
o2_a = 99.26489671
abs diff = 6.356117330000004
o1_a = 97.58130247
o2_a = 103.9374198
abs diff = 3.500000019585059e-06
o1_a = 208.0306983
o2_a = 208.0306948
abs diff = 1.3000000080864993e-06
o1_a = 101.1242598
o2_a = 101.1242585
abs diff = 1.900000000887303e-06
o1_a = 109.0858324
o2_a = 109.0858305
abs diff = 3.6000000136482413e-06
o1_a = 216.506569
o2_a = 216.5065654
abs diff = 6.765574480000012
o1_a = 96.61631442
o2_a = 103.3818889
abs diff = 2.00000000916134e-06
o1_a = 112.7029437
o2_a = 112.7029417
abs diff = 35.15611039999999
o1_a = 189.2148152
o2_a = 224.3709256
abs diff = 0.07739030000000469
o1_a = 105.5286012
o2_a = 105.6059915
abs diff = 0.6433114000000018
o1_a = 114.533574
o2_a = 115.1768854
abs diff = 3.999999989900971e-06
o1_a = 232.0485297
o2_a = 232.0485257
abs diff = 0.035510700000003226
o1_a = 105.9360799
o2_a = 105.9715906
abs diff = 0.12516389999998978
o1_a = 113.9784544
o2_a = 114.1036183
abs diff = 4.100000012385863e-06
o1_a = 239.5270447
o2_a = 239.5270406
abs diff = 1.2999999938756446e-06
o1_a = 105.7862313
o2_a = 105.78623
abs diff = 1.900000000887303e-06
o1_a = 108.52932
o2_a = 108.5293181
abs diff = 4.300000000512227e-06
o1_a = 246.7943387
o2_a = 246.7943344
abs diff = 3.429252500000004
o1_a = 101.9571728
o2_a = 105.3864253
abs diff = 10.306834219999999
o1_a = 93.67234608
o2_a = 103.9791803
abs diff = 48.606040199999995
o1_a = 192.5584688
o2_a = 241.164509
abs diff = 0.7479256000000021
o1_a = 104.0496174
o2_a = 104.797543
abs diff = 0.13694050000000857
o1_a = 100.5018749
o2_a = 100.3649344
abs diff = 1.295027800000014
o1_a = 233.2255318
o2_a = 234.5205596
abs diff = 1.3000000080864993e-06
o1_a = 104.0452848
o2_a = 104.0452835
abs diff = 1.7100000064829146e-06
o1_a = 97.60373907
o2_a = 97.60373736
abs diff = 3.8999999958377884e-06
o1_a = 228.1167706
o2_a = 228.1167667
abs diff = 1.465846600000006
o1_a = 101.6882029
o2_a = 103.1540495
abs diff = 0.6196199500000006
o1_a = 94.99499951
o2_a = 95.61461946
abs diff = 3.800000001774606e-06
o1_a = 221.96397
o2_a = 221.9639662
abs diff = 0.47177660000001254
o1_a = 101.6750472
o2_a = 102.1468238
abs diff = 0.8246377699999954
o1_a = 95.14145185
o2_a = 94.31681408
abs diff = 3.7000000077114237e-06
o1_a = 216.0726505
o2_a = 216.0726468
abs diff = 1.2999999938756446e-06
o1_a = 101.0461792
o2_a = 101.0461779
abs diff = 1.6199999919308539e-06
o1_a = 93.62980058
o2_a = 93.62979896
abs diff = 3.6000000136482413e-06
o1_a = 210.4528821
o2_a = 210.4528785
abs diff = 0.4272120900000118
o1_a = 99.45338394
o2_a = 99.88059603
abs diff = 1.026236659999995
o1_a = 92.44744865
o2_a = 93.47368531
abs diff = 3.4999999911633495e-06
o1_a = 205.1143376
o2_a = 205.1143341
abs diff = 1.1999999998124622e-06
o1_a = 100.2835727
o2_a = 100.2835715
abs diff = 1.6000000044869012e-06
o1_a = 93.88468851
o2_a = 93.88468691
abs diff = 3.399999997100167e-06
o1_a = 200.0646106
o2_a = 200.0646072
abs diff = 1.2999999938756446e-06
o1_a = 100.624758
o2_a = 100.6247567
abs diff = 1.6199999919308539e-06
o1_a = 94.85467109
o2_a = 94.85466947
abs diff = 3.3000000030369847e-06
o1_a = 195.7734237
o2_a = 195.7734204
abs diff = 1.1999999998124622e-06
o1_a = 100.8385694
o2_a = 100.8385682
abs diff = 1.629999999863685e-06
o1_a = 95.46221541
o2_a = 95.46221378
abs diff = 3.3000000030369847e-06
o1_a = 193.079588
o2_a = 193.0795847
o1_a [127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 165.5946429, 117.9955882, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 26.92126434, 113.6637821, 127.6893051, 0.0, 10.09163902, 189.2148152, 111.7693223, 0.0, 32.66883179, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 107.1333087, 0.0, 37.87986154, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 127.6893051, 0.0, 10.09163902, 189.2148152, 103.7696277, 0.0, 41.79530516, 189.2148152, 102.553217, 0.0, 43.18195142, 189.2148152, 116.9090289, 0.0, 16.15190959, 166.311555, 101.0007594, 0.0, 40.08192817, 182.4046244, 100.3220682, 0.0, 43.48069611, 189.2148152, 111.7193699, 0.0, 31.35523359, 173.2208279, 100.2331125, 0.0, 39.30940423, 189.2148152, 98.60919826, 0.0, 43.72430542, 189.2148152, 106.7035688, 0.0, 37.55134452, 189.2148152, 102.8534912, 0.0, 41.25287517, 189.2148152, 97.22892325, 0.0, 45.97926742, 189.2148152, 101.8562192, 0.0, 43.15867651, 189.2148152, 99.66137081, 0.0, 45.67601704, 189.2148152, 96.04936185, 0.0, 51.33889298, 189.2148152, 95.55972991, 0.0, 53.82854337, 189.2148152, 98.20586034, 0.0, 44.33328357, 189.2148152, 94.48649624, 0.0, 54.21761271, 189.2148152, 93.73347788, 0.0, 61.00365755, 189.2148152, 97.26050969, 0.0, 46.47652073, 189.2148152, 95.28116739, 0.0, 53.44000991, 189.2148152, 92.65744635, 0.0, 68.76526834, 189.2148152, 93.60636027, 0.0, 71.50389607, 189.2148152, 93.25719057, 0.0, 66.33299975, 189.2148152, 94.95720506, 0.0, 76.41706903, 189.2148152, 95.91438361, 0.0, 84.40451554, 189.2148152, 93.38085625, 0.0, 70.37478769, 189.2148152, 97.52694512, 0.0, 97.58130247, 208.0306983, 101.1242598, 0.0, 109.0858324, 216.506569, 96.61631442, 0.0, 112.7029437, 189.2148152, 105.5286012, 0.0, 114.533574, 232.0485297, 105.9360799, 0.0, 113.9784544, 239.5270447, 105.7862313, 0.0, 108.52932, 246.7943387, 101.9571728, 0.0, 93.67234608, 192.5584688, 104.0496174, 0.0, 100.5018749, 233.2255318, 104.0452848, 0.0, 97.60373907, 228.1167706, 101.6882029, 0.0, 94.99499951, 221.96397, 101.6750472, 0.0, 95.14145185, 216.0726505, 101.0461792, 0.0, 93.62980058, 210.4528821, 99.45338394, 0.0, 92.44744865, 205.1143376, 100.2835727, 0.0, 93.88468851, 200.0646106, 100.624758, 0.0, 94.85467109, 195.7734237, 100.8385694, 0.0, 95.46221541, 193.079588]
o2_a [127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 127.6893034, 0.0, 10.09163899, 189.2148121, 122.9464469, 0.0, 10.09163899, 165.5946401, 117.9426934, 0.0, 19.25012189, 119.8840089, 114.8802829, 0.0, 26.921264, 113.6637802, 113.1554513, 0.0, 31.04083312, 145.4571544, 111.7693209, 0.0, 33.68251764, 175.9833623, 110.1957694, 0.0, 35.4200803, 189.2148121, 108.539878, 0.0, 36.66019517, 189.2148121, 107.1333073, 0.0, 37.87986099, 189.2148121, 105.9984125, 0.0, 39.21826875, 189.2148121, 104.9156704, 0.0, 40.62993126, 189.2148121, 103.7696264, 0.0, 41.96829401, 189.2148121, 102.5532157, 0.0, 43.18195077, 189.2148121, 101.6242621, 0.0, 43.63787878, 189.2148121, 100.9567007, 0.0, 43.55188875, 189.2148121, 100.322067, 0.0, 43.48069545, 189.2148121, 99.72760295, 0.0, 43.43314677, 189.2148121, 99.14733722, 0.0, 43.52191497, 189.2148121, 98.60919703, 0.0, 43.72430476, 189.2148121, 98.09329141, 0.0, 44.16126157, 189.2148121, 97.63370132, 0.0, 44.88678879, 189.2148121, 97.22892204, 0.0, 45.97926672, 189.2148121, 96.83241494, 0.0, 47.43768636, 189.2148121, 96.44683108, 0.0, 49.20700911, 189.2148121, 96.04936066, 0.0, 51.33889218, 189.2148121, 95.55972873, 0.0, 53.82854252, 189.2148121, 95.01029712, 0.0, 56.35302718, 189.2148121, 94.43023958, 0.0, 58.64327803, 189.2148121, 93.73347672, 0.0, 61.00365657, 189.2148121, 93.16994132, 0.0, 63.51318309, 189.2148121, 92.91805654, 0.0, 66.08881905, 189.2148121, 92.65744522, 0.0, 68.76526722, 189.2148121, 93.60635913, 0.0, 71.50389489, 189.2148121, 94.27347437, 0.0, 74.17944527, 189.2148121, 94.96643113, 0.0, 76.55421196, 189.2148121, 95.91438243, 0.0, 84.40451411, 189.2148121, 97.46993316, 0.0, 95.62556497, 198.0068374, 99.26489671, 0.0, 103.9374198, 208.0306948, 101.1242585, 0.0, 109.0858305, 216.5065654, 103.3818889, 0.0, 112.7029417, 224.3709256, 105.6059915, 0.0, 115.1768854, 232.0485257, 105.9715906, 0.0, 114.1036183, 239.5270406, 105.78623, 0.0, 108.5293181, 246.7943344, 105.3864253, 0.0, 103.9791803, 241.164509, 104.797543, 0.0, 100.3649344, 234.5205596, 104.0452835, 0.0, 97.60373736, 228.1167667, 103.1540495, 0.0, 95.61461946, 221.9639662, 102.1468238, 0.0, 94.31681408, 216.0726468, 101.0461779, 0.0, 93.62979896, 210.4528785, 99.88059603, 0.0, 93.47368531, 205.1143341, 100.2835715, 0.0, 93.88468691, 200.0646072, 100.6247567, 0.0, 94.85466947, 195.7734204, 100.8385682, 0.0, 95.46221378, 193.0795847]
L1 rel_error 3.108405262296241
L2 rel_error 0.576228090966141
Linf rel_error 0.2809254323466376
final pass array = [False]
Traceback (most recent call last):
  File "compare_mam4xx_mam4.py", line 136, in <module>
    assert(np.all(pass_all_tests))
AssertionError

```

File: src/mam4xx/mo_photo.hpp [hunk 1]
```diff
@@ -750,19 +756,21 @@ void table_photo(const View2D &photo, // out
...
          ... compute eff_alb and cld_mult -- needs to be before jlong
     -----------------------------------------------------------------*/
-    cloud_mod(zen_angle, clouds.data(), lwc.data(), pdel.data(),
+    cloud_mod(zen_angle, clouds, lwc, pdel,
               srf_alb, //  in
               eff_alb, cld_mult);
-
-    for (int kk = 0; kk < pver; ++kk) {
+    for (int kk = 0; kk < pver; kk++) {
+      // Kokkos::parallel_for(Kokkos::TeamVectorRange(team, pver_local),
```
Comment (singhbalwinder) [inline src/mam4xx/mo_photo.hpp:763]: Can we remove all the commented-out code?

**Review by singhbalwinder — APPROVED** [review]
Looks good! Thanks for fixing this issue. If all the tests are passing let's merge it.

**Review by jeff-cohere — APPROVED** [review]
👍

**Comment by odiazib** [comment]
> Looks good! Thanks for fixing this issue. If all the tests are passing let's merge it.

All tests are passing. I am waiting for all tests to run again to double-check that this is still true.

**Comment by odiazib** [comment]
All tests are passing. I am going to merge this branch.

**Pull request merged**